src/Compiler/Driver/ParseAndCheckInputs.fs

```diff
@@ -1961,7 +1961,54 @@ let CheckMultipleInputsUsingGraphMode
 
         partialResults, tcState)
 
+let TryReuseTypecheckingResults (tcConfig: TcConfig) inputs =
+    let tcDataFileName = FileSystem.GetFullPathShim "FSharpTypecheckingData"
+    let nl = Environment.NewLine
+
+    let sourceFiles =
+        inputs
+        |> Seq.toArray
+        |> Array.mapi (fun idx (input: ParsedInput) ->
+            {
+                Idx = idx
+                FileName = input.FileName
+                ParsedInput = input
```

Comment:
In general I am missing "lastupdated" timestamps for making the comparison relevant.
Also compiler version or any other header to be a quick circuit breaker in case of newer compiler getting access to older files.

(might even be version/timestamp/.. of Fsharp.Compiler.Service.dll itself I guess?)

Reply:
Correct, I added a notion of this to what we write to the TcData.